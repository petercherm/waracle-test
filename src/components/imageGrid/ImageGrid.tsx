import { ImageType } from "../../models/theCatApi/image.models";
import { FetchStatus, UploadStatus } from "../../store/rootAction";
import { AddImage } from "../addImage/AddImage";
import { ImageTile } from "../imageTile/ImageTile";
import { ErrorMessage, ImageGridContainer } from "./ImageGrid.styles";

export interface ImageGridProps {
  images: ImageType[];
  fetchStatus: FetchStatus;
  uploadStatus: UploadStatus;
  onUploadImage: () => void;
}

export const ImageGrid = ({
  images,
  fetchStatus,
  uploadStatus,
  onUploadImage
}: ImageGridProps) => {
  const { isFetching, isError, error } = fetchStatus;
  const { isError: isUploadError, error: uploadError } = uploadStatus;

  if (isError) {
    return (
      <ErrorMessage>
        Something went wrong: <strong>{error}</strong>
      </ErrorMessage>
    );
  }

  if (isFetching) {
    return <p>Loading images...</p>;
  }
  const renderImages = () =>
    images.map((image: ImageType) => (
      <ImageTile key={image.id} image={image} />
    ));

  return (
    <>
      {isUploadError && (
        <ErrorMessage>
          Image cound not be uploaded: <strong>{uploadError}</strong>
        </ErrorMessage>
      )}
      <p>
        {images.length
          ? `Your cats - number of images: ${images.length}`
          : "It looks like you haven't uploaded any images yet"}
      </p>
      <ImageGridContainer>
        {renderImages()}
        <AddImage onUploadImage={onUploadImage} />
      </ImageGridContainer>
    </>
  );
};
