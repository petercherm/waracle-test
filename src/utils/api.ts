import { AxiosError } from "axios";

export interface ErrorResponseMessage {
  message: string;
}

export const returnError = (
  error: AxiosError<ErrorResponseMessage, unknown>
) => {
  if (error.response) {
    return {
      error: { message: error.response.data?.message || "Unknown error" }
    };
  }
  return { error };
};
