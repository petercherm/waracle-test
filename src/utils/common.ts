const isObject = (item: any): boolean => {
  return item && typeof item === "object" && !Array.isArray(item);
};

export const mergeDeep = <T, S>(target: T, ...sources: S[]): T => {
  if (!sources.length) return target;
  const source = sources.shift();

  if (isObject(target) && isObject(source)) {
    for (const key in source) {
      if (isObject(source[key])) {
        if (!(target as Record<string, unknown>)[key])
          Object.assign(target, {
            [key]: {}
          });
        mergeDeep(
          (target as Record<string, any>)[key],
          (source as Record<string, any>)[key]
        );
      } else {
        Object.assign(target, {
          [key]: source[key]
        });
      }
    }
  }

  return mergeDeep(target, ...sources);
};
