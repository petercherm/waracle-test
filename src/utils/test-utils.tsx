/* eslint-disable import/export */
import { rootReducer } from "../store/rootReducer";
import { render, RenderResult } from "@testing-library/react";
import { FunctionComponent } from "react";
import { Provider } from "react-redux";
import { createStore } from "redux";
import { ThemeProvider } from "styled-components";
import { baseTheme } from "../theme/baseTheme";

const customRender = (
  ui: JSX.Element,
  {
    initialState = {},
    store = createStore(rootReducer, initialState),
    ...renderOptions
  }: any = {}
): RenderResult => {
  const Wrapper: FunctionComponent = ({ children }) => {
    return (
      <Provider store={store}>
        <ThemeProvider theme={baseTheme}>{children}</ThemeProvider>
      </Provider>
    );
  };
  return render(ui, { wrapper: Wrapper, ...renderOptions });
};

// re-export everything
export * from "@testing-library/react";

// override render method
export { customRender as render };
