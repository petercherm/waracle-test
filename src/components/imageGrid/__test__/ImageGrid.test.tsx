import "@testing-library/jest-dom/extend-expect";
import { ImageType } from "../../../models/theCatApi/image.models";
import { render } from "../../../utils/test-utils";
import { ImageGrid, ImageGridProps } from "../ImageGrid";

describe("GIVEN ImageGrid component", () => {
  const images: ImageType[] = [
    {
      breed_ids: null,
      breeds: [],
      created_at: "2021-10-15T01:17:52.000Z",
      height: 3024,
      id: "geQomtavn",
      original_filename: "F228D6F8-2E6D-4E43-85EE-4275CD9E75A3.jpeg",
      sub_id: "demo-4e5577",
      url: "https://cdn2.thecatapi.com/images/geQomtavn.jpg",
      width: 4032
    },
    {
      breed_ids: null,
      breeds: [],
      created_at: "2021-10-15T01:17:53.000Z",
      height: 3024,
      id: "abCDfe",
      original_filename: "F228D6F8-2E6D-4E43-85EE-4275CD9E75A4.jpeg",
      sub_id: "demo-4e5577",
      url: "https://cdn2.thecatapi.com/images/geQomtavn.jpg",
      width: 4032
    }
  ];

  const props: ImageGridProps = {
    images,
    fetchStatus: {
      isError: false,
      isFetching: false,
      error: ""
    },
    uploadStatus: {
      isError: false,
      isUploading: false,
      error: ""
    },
    onUploadImage: jest.fn(),
    onViewImage: jest.fn()
  };

  test("it matches the snapshot", () => {
    const container = render(<ImageGrid {...props} />);

    expect(container).toMatchSnapshot();
  });

  test("it renders all the elements correctly", () => {
    const { getAllByTestId, queryByTestId } = render(<ImageGrid {...props} />);

    expect(queryByTestId("imageGrid__error--uploadError")).toBeNull();
    expect(queryByTestId("imageGrid__loading")).toBeFalsy();
    expect(queryByTestId("imageGrid__error--fetchError")).toBeFalsy();
    expect(queryByTestId("addImage__container")).toBeTruthy();
    expect(getAllByTestId("imageTile__container").length).toEqual(2);
  });

  test("it renders the fetch error if the fetch status shows error", () => {
    const errorProps = {
      ...props,
      fetchStatus: {
        ...props.fetchStatus,
        isError: true,
        error: "Error"
      }
    };
    const { queryByTestId } = render(<ImageGrid {...errorProps} />);

    expect(queryByTestId("imageGrid__error--uploadError")).toBeNull();
    expect(queryByTestId("imageGrid__loading")).toBeFalsy();
    expect(queryByTestId("imageGrid__error--fetchError")).toBeTruthy();
    expect(queryByTestId("addImage__container")).toBeFalsy();
    expect(queryByTestId("imageTile__container")).toBeNull();
  });

  test("it renders the fetch error if the upload status shows error", () => {
    const errorProps = {
      ...props,
      uploadStatus: {
        ...props.uploadStatus,
        isError: true,
        error: "Error"
      }
    };
    const { queryByTestId, queryAllByTestId } = render(
      <ImageGrid {...errorProps} />
    );

    expect(queryByTestId("imageGrid__error--uploadError")).toBeTruthy();
    expect(queryByTestId("imageGrid__loading")).toBeFalsy();
    expect(queryByTestId("imageGrid__error--fetchError")).toBeFalsy();
    expect(queryByTestId("addImage__container")).toBeTruthy();
    expect(queryAllByTestId("imageTile__container").length).toEqual(2);
  });

  test("it renders the loading message if the fetching is in progress", () => {
    const loadingProps = {
      ...props,
      fetchStatus: {
        ...props.fetchStatus,
        isFetching: true
      }
    };
    const { queryByTestId } = render(<ImageGrid {...loadingProps} />);

    expect(queryByTestId("imageGrid__error--uploadError")).toBeFalsy();
    expect(queryByTestId("imageGrid__loading")).toBeTruthy();
    expect(queryByTestId("imageGrid__error--fetchError")).toBeFalsy();
    expect(queryByTestId("addImage__container")).toBeFalsy();
    expect(queryByTestId("imageTile__container")).toBeFalsy();
  });
});
