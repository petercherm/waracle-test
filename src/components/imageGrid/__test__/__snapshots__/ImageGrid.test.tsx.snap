// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GIVEN ImageGrid component it matches the snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <p>
        Your cats - number of images: 2
      </p>
      <ul
        class="sc-jrAFXE iLimCB"
      >
        <li
          class="sc-jSgvzq gssyfx"
          data-testid="imageTile__container"
        >
          <div
            class="sc-gKseQn jwVoyy"
          >
            <div
              class="sc-iBPTik gLaBuZ"
            >
              <div
                class="sc-fubCzh WLQNF"
                role="button"
              />
            </div>
            <div
              class="sc-pGacB ljFrZB"
            >
              <div
                class="sc-dlfnuX hZvLBn"
                role="button"
              >
                <svg
                  clip-rule="evenodd"
                  stroke-miterlimit="1.5"
                  viewBox="0 0 50 50"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M25 13c4.21-8 12.631-8 16.842-4 4.211 4 4.211 12 0 20C38.894 35 31.315 41 25 45c-6.316-4-13.895-10-16.842-16-4.21-8-4.21-16 0-20S20.79 5 25 13Z"
                    fill="#eee"
                    stroke="black"
                    stroke-width="2"
                  />
                </svg>
              </div>
              <div
                class="sc-eCstlR hGtvQl"
              >
                <div
                  class="sc-hKgJUU cGeifQ"
                  role="button"
                >
                  <svg
                    clip-rule="evenodd"
                    fill-rule="evenodd"
                    stroke-linejoin="round"
                    stroke-miterlimit="2"
                    viewBox="0 0 50 50"
                    xml:space="preserve"
                  >
                    <path
                      d="M40.04 21.667h-9.4l3.378-10.976c.711-2.289-1.155-4.577-3.733-4.577-1.288 0-2.533.533-3.377 1.444l-13.02 14.11H5v22.218h32.062c2.355 0 4.4-1.488 4.866-3.577l2.977-13.332c.6-2.755-1.733-5.31-4.865-5.31ZM13.888 41.664H7.222V23.89h6.666v17.775Zm28.84-15.175L39.751 39.82c-.223 1.067-1.356 1.844-2.689 1.844H16.11v-19.13L28.552 9.069c.422-.467 1.067-.733 1.733-.733.578 0 1.111.244 1.4.666.156.222.333.578.2 1.045l-3.377 10.976-.889 2.866h12.398c.911 0 1.778.378 2.289 1.022.289.333.578.889.422 1.578Z"
                      fill="green"
                      fill-rule="nonzero"
                    />
                  </svg>
                </div>
                <span>
                  0
                </span>
                <div
                  class="sc-hKgJUU jOGJDV"
                  role="button"
                >
                  <svg
                    clip-rule="evenodd"
                    fill-rule="evenodd"
                    stroke-linejoin="round"
                    stroke-miterlimit="2"
                    viewBox="0 0 50 50"
                    xml:space="preserve"
                  >
                    <path
                      d="M40.04 21.667h-9.4l3.378-10.976c.711-2.289-1.155-4.577-3.733-4.577-1.288 0-2.533.533-3.377 1.444l-13.02 14.11H5v22.218h32.062c2.355 0 4.4-1.488 4.866-3.577l2.977-13.332c.6-2.755-1.733-5.31-4.865-5.31ZM13.888 41.664H7.222V23.89h6.666v17.775Zm28.84-15.175L39.751 39.82c-.223 1.067-1.356 1.844-2.689 1.844H16.11v-19.13L28.552 9.069c.422-.467 1.067-.733 1.733-.733.578 0 1.111.244 1.4.666.156.222.333.578.2 1.045l-3.377 10.976-.889 2.866h12.398c.911 0 1.778.378 2.289 1.022.289.333.578.889.422 1.578Z"
                      fill="red"
                      fill-rule="nonzero"
                    />
                  </svg>
                </div>
              </div>
            </div>
          </div>
        </li>
        <li
          class="sc-jSgvzq gssyfx"
          data-testid="imageTile__container"
        >
          <div
            class="sc-gKseQn jwVoyy"
          >
            <div
              class="sc-iBPTik gLaBuZ"
            >
              <div
                class="sc-fubCzh WLQNF"
                role="button"
              />
            </div>
            <div
              class="sc-pGacB ljFrZB"
            >
              <div
                class="sc-dlfnuX hZvLBn"
                role="button"
              >
                <svg
                  clip-rule="evenodd"
                  stroke-miterlimit="1.5"
                  viewBox="0 0 50 50"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M25 13c4.21-8 12.631-8 16.842-4 4.211 4 4.211 12 0 20C38.894 35 31.315 41 25 45c-6.316-4-13.895-10-16.842-16-4.21-8-4.21-16 0-20S20.79 5 25 13Z"
                    fill="#eee"
                    stroke="black"
                    stroke-width="2"
                  />
                </svg>
              </div>
              <div
                class="sc-eCstlR hGtvQl"
              >
                <div
                  class="sc-hKgJUU cGeifQ"
                  role="button"
                >
                  <svg
                    clip-rule="evenodd"
                    fill-rule="evenodd"
                    stroke-linejoin="round"
                    stroke-miterlimit="2"
                    viewBox="0 0 50 50"
                    xml:space="preserve"
                  >
                    <path
                      d="M40.04 21.667h-9.4l3.378-10.976c.711-2.289-1.155-4.577-3.733-4.577-1.288 0-2.533.533-3.377 1.444l-13.02 14.11H5v22.218h32.062c2.355 0 4.4-1.488 4.866-3.577l2.977-13.332c.6-2.755-1.733-5.31-4.865-5.31ZM13.888 41.664H7.222V23.89h6.666v17.775Zm28.84-15.175L39.751 39.82c-.223 1.067-1.356 1.844-2.689 1.844H16.11v-19.13L28.552 9.069c.422-.467 1.067-.733 1.733-.733.578 0 1.111.244 1.4.666.156.222.333.578.2 1.045l-3.377 10.976-.889 2.866h12.398c.911 0 1.778.378 2.289 1.022.289.333.578.889.422 1.578Z"
                      fill="green"
                      fill-rule="nonzero"
                    />
                  </svg>
                </div>
                <span>
                  0
                </span>
                <div
                  class="sc-hKgJUU jOGJDV"
                  role="button"
                >
                  <svg
                    clip-rule="evenodd"
                    fill-rule="evenodd"
                    stroke-linejoin="round"
                    stroke-miterlimit="2"
                    viewBox="0 0 50 50"
                    xml:space="preserve"
                  >
                    <path
                      d="M40.04 21.667h-9.4l3.378-10.976c.711-2.289-1.155-4.577-3.733-4.577-1.288 0-2.533.533-3.377 1.444l-13.02 14.11H5v22.218h32.062c2.355 0 4.4-1.488 4.866-3.577l2.977-13.332c.6-2.755-1.733-5.31-4.865-5.31ZM13.888 41.664H7.222V23.89h6.666v17.775Zm28.84-15.175L39.751 39.82c-.223 1.067-1.356 1.844-2.689 1.844H16.11v-19.13L28.552 9.069c.422-.467 1.067-.733 1.733-.733.578 0 1.111.244 1.4.666.156.222.333.578.2 1.045l-3.377 10.976-.889 2.866h12.398c.911 0 1.778.378 2.289 1.022.289.333.578.889.422 1.578Z"
                      fill="red"
                      fill-rule="nonzero"
                    />
                  </svg>
                </div>
              </div>
            </div>
          </div>
        </li>
        <li
          class="sc-bdfBQB dzUwvV"
          data-testid="addImage__container"
        >
          <div
            class="sc-gsTEea fZByWz"
            role="button"
          />
        </li>
      </ul>
    </div>
  </body>,
  "container": <div>
    <p>
      Your cats - number of images: 2
    </p>
    <ul
      class="sc-jrAFXE iLimCB"
    >
      <li
        class="sc-jSgvzq gssyfx"
        data-testid="imageTile__container"
      >
        <div
          class="sc-gKseQn jwVoyy"
        >
          <div
            class="sc-iBPTik gLaBuZ"
          >
            <div
              class="sc-fubCzh WLQNF"
              role="button"
            />
          </div>
          <div
            class="sc-pGacB ljFrZB"
          >
            <div
              class="sc-dlfnuX hZvLBn"
              role="button"
            >
              <svg
                clip-rule="evenodd"
                stroke-miterlimit="1.5"
                viewBox="0 0 50 50"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M25 13c4.21-8 12.631-8 16.842-4 4.211 4 4.211 12 0 20C38.894 35 31.315 41 25 45c-6.316-4-13.895-10-16.842-16-4.21-8-4.21-16 0-20S20.79 5 25 13Z"
                  fill="#eee"
                  stroke="black"
                  stroke-width="2"
                />
              </svg>
            </div>
            <div
              class="sc-eCstlR hGtvQl"
            >
              <div
                class="sc-hKgJUU cGeifQ"
                role="button"
              >
                <svg
                  clip-rule="evenodd"
                  fill-rule="evenodd"
                  stroke-linejoin="round"
                  stroke-miterlimit="2"
                  viewBox="0 0 50 50"
                  xml:space="preserve"
                >
                  <path
                    d="M40.04 21.667h-9.4l3.378-10.976c.711-2.289-1.155-4.577-3.733-4.577-1.288 0-2.533.533-3.377 1.444l-13.02 14.11H5v22.218h32.062c2.355 0 4.4-1.488 4.866-3.577l2.977-13.332c.6-2.755-1.733-5.31-4.865-5.31ZM13.888 41.664H7.222V23.89h6.666v17.775Zm28.84-15.175L39.751 39.82c-.223 1.067-1.356 1.844-2.689 1.844H16.11v-19.13L28.552 9.069c.422-.467 1.067-.733 1.733-.733.578 0 1.111.244 1.4.666.156.222.333.578.2 1.045l-3.377 10.976-.889 2.866h12.398c.911 0 1.778.378 2.289 1.022.289.333.578.889.422 1.578Z"
                    fill="green"
                    fill-rule="nonzero"
                  />
                </svg>
              </div>
              <span>
                0
              </span>
              <div
                class="sc-hKgJUU jOGJDV"
                role="button"
              >
                <svg
                  clip-rule="evenodd"
                  fill-rule="evenodd"
                  stroke-linejoin="round"
                  stroke-miterlimit="2"
                  viewBox="0 0 50 50"
                  xml:space="preserve"
                >
                  <path
                    d="M40.04 21.667h-9.4l3.378-10.976c.711-2.289-1.155-4.577-3.733-4.577-1.288 0-2.533.533-3.377 1.444l-13.02 14.11H5v22.218h32.062c2.355 0 4.4-1.488 4.866-3.577l2.977-13.332c.6-2.755-1.733-5.31-4.865-5.31ZM13.888 41.664H7.222V23.89h6.666v17.775Zm28.84-15.175L39.751 39.82c-.223 1.067-1.356 1.844-2.689 1.844H16.11v-19.13L28.552 9.069c.422-.467 1.067-.733 1.733-.733.578 0 1.111.244 1.4.666.156.222.333.578.2 1.045l-3.377 10.976-.889 2.866h12.398c.911 0 1.778.378 2.289 1.022.289.333.578.889.422 1.578Z"
                    fill="red"
                    fill-rule="nonzero"
                  />
                </svg>
              </div>
            </div>
          </div>
        </div>
      </li>
      <li
        class="sc-jSgvzq gssyfx"
        data-testid="imageTile__container"
      >
        <div
          class="sc-gKseQn jwVoyy"
        >
          <div
            class="sc-iBPTik gLaBuZ"
          >
            <div
              class="sc-fubCzh WLQNF"
              role="button"
            />
          </div>
          <div
            class="sc-pGacB ljFrZB"
          >
            <div
              class="sc-dlfnuX hZvLBn"
              role="button"
            >
              <svg
                clip-rule="evenodd"
                stroke-miterlimit="1.5"
                viewBox="0 0 50 50"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M25 13c4.21-8 12.631-8 16.842-4 4.211 4 4.211 12 0 20C38.894 35 31.315 41 25 45c-6.316-4-13.895-10-16.842-16-4.21-8-4.21-16 0-20S20.79 5 25 13Z"
                  fill="#eee"
                  stroke="black"
                  stroke-width="2"
                />
              </svg>
            </div>
            <div
              class="sc-eCstlR hGtvQl"
            >
              <div
                class="sc-hKgJUU cGeifQ"
                role="button"
              >
                <svg
                  clip-rule="evenodd"
                  fill-rule="evenodd"
                  stroke-linejoin="round"
                  stroke-miterlimit="2"
                  viewBox="0 0 50 50"
                  xml:space="preserve"
                >
                  <path
                    d="M40.04 21.667h-9.4l3.378-10.976c.711-2.289-1.155-4.577-3.733-4.577-1.288 0-2.533.533-3.377 1.444l-13.02 14.11H5v22.218h32.062c2.355 0 4.4-1.488 4.866-3.577l2.977-13.332c.6-2.755-1.733-5.31-4.865-5.31ZM13.888 41.664H7.222V23.89h6.666v17.775Zm28.84-15.175L39.751 39.82c-.223 1.067-1.356 1.844-2.689 1.844H16.11v-19.13L28.552 9.069c.422-.467 1.067-.733 1.733-.733.578 0 1.111.244 1.4.666.156.222.333.578.2 1.045l-3.377 10.976-.889 2.866h12.398c.911 0 1.778.378 2.289 1.022.289.333.578.889.422 1.578Z"
                    fill="green"
                    fill-rule="nonzero"
                  />
                </svg>
              </div>
              <span>
                0
              </span>
              <div
                class="sc-hKgJUU jOGJDV"
                role="button"
              >
                <svg
                  clip-rule="evenodd"
                  fill-rule="evenodd"
                  stroke-linejoin="round"
                  stroke-miterlimit="2"
                  viewBox="0 0 50 50"
                  xml:space="preserve"
                >
                  <path
                    d="M40.04 21.667h-9.4l3.378-10.976c.711-2.289-1.155-4.577-3.733-4.577-1.288 0-2.533.533-3.377 1.444l-13.02 14.11H5v22.218h32.062c2.355 0 4.4-1.488 4.866-3.577l2.977-13.332c.6-2.755-1.733-5.31-4.865-5.31ZM13.888 41.664H7.222V23.89h6.666v17.775Zm28.84-15.175L39.751 39.82c-.223 1.067-1.356 1.844-2.689 1.844H16.11v-19.13L28.552 9.069c.422-.467 1.067-.733 1.733-.733.578 0 1.111.244 1.4.666.156.222.333.578.2 1.045l-3.377 10.976-.889 2.866h12.398c.911 0 1.778.378 2.289 1.022.289.333.578.889.422 1.578Z"
                    fill="red"
                    fill-rule="nonzero"
                  />
                </svg>
              </div>
            </div>
          </div>
        </div>
      </li>
      <li
        class="sc-bdfBQB dzUwvV"
        data-testid="addImage__container"
      >
        <div
          class="sc-gsTEea fZByWz"
          role="button"
        />
      </li>
    </ul>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
